crates/pallets/starknet/src/tests/deploy_account_tx.rs

```diff
@@ -262,14 +262,12 @@ fn given_contract_run_deploy_account_braavos_tx_works() {
             class_hash: proxy_class_hash.into(),
         };
 
-        // Braavos has a complicated signature mecanism, they add stuffs around the tx_hash and then sign
-        // the whole thing. This hardcoded value is the expected "thing" that bravos expect you to
-        // sign for this transaction. Roll with it for now
-        let value_to_sign =
-            Felt252Wrapper::from_hex_be("0x06a8bb3d81c2ad23db93f01f72f987feac5210a95bc530eabb6abfaa5a769944").unwrap();
-        let mut signatures = sign_message_hash(value_to_sign);
-        signatures.extend_from_slice(&[Felt252Wrapper::ZERO; 8]);
-        deploy_tx.signature = signatures;
+        let tx_hash = deploy_tx.compute_hash::<<MockRuntime as Config>::SystemHash>(Starknet::chain_id(), false);
+        deploy_tx.signature = sign_message_hash_braavos::<<MockRuntime as Config>::SystemHash>(
```

Comment:
2 small details:
1. We should probably not use the `SystemHash` here but directly the right hash for this braavos contract.
2. I prefer that we move the signature computation above the `deploy_tx` declaration with something like `let signature = sign...` then just fill it in the `deploy_tx` struct.

Reply:
Could you please clarify your second suggestion? It seems like signing a hash before obtaining the hash itself might not be possible.